src/main/java/listeners/RoomListener.java

```diff
@@ -1,17 +1,18 @@
 package listeners;
 
 import model.InboundMessage;
+import model.Initiator;
 import model.Stream;
 import model.events.*;
 
 public interface RoomListener extends DatafeedListener {
-    void onRoomMessage(InboundMessage message);
-    void onRoomCreated(RoomCreated roomCreated);
-    void onRoomDeactivated(RoomDeactivated roomDeactivated);
-    void onRoomMemberDemotedFromOwner(RoomMemberDemotedFromOwner roomMemberDemotedFromOwner);
-    void onRoomMemberPromotedToOwner(RoomMemberPromotedToOwner roomMemberPromotedToOwner);
-    void onRoomReactivated(Stream stream);
-    void onRoomUpdated(RoomUpdated roomUpdated);
-    void onUserJoinedRoom(UserJoinedRoom userJoinedRoom);
-    void onUserLeftRoom(UserLeftRoom userLeftRoom);
+    void onRoomMessage(InboundMessage message, Initiator initiator);
+    void onRoomCreated(RoomCreated roomCreated, Initiator initiator);
+    void onRoomDeactivated(RoomDeactivated roomDeactivated, Initiator initiator);
+    void onRoomMemberDemotedFromOwner(RoomMemberDemotedFromOwner roomMemberDemotedFromOwner, Initiator initiator);
+    void onRoomMemberPromotedToOwner(RoomMemberPromotedToOwner roomMemberPromotedToOwner, Initiator initiator);
+    void onRoomReactivated(Stream stream, Initiator initiator);
+    void onRoomUpdated(RoomUpdated roomUpdated, Initiator initiator);
+    void onUserJoinedRoom(UserJoinedRoom userJoinedRoom, Initiator initiator);
+    void onUserLeftRoom(UserLeftRoom userLeftRoom, Initiator initiator);
```

Comment:
Here you introduced a breaking in the API. We have to be very careful with that in our following developments! 

2 solutions : 
- mark existing method as deprecated and duplicate it with second `Initiator` param
- use varargs for `Initiator` param and check in the method if it's present or not : 

```java
void onRoomMessage(InboundMessage message, Initiator... initiator);
```

Reply:
(I actually prefer the first solution, `@Deprecated` + second method with `Initiator` param)

Reply:
Since the customer only requested for userjoinedroom event, I have an idea that we can create an Initiator field in UserJoinedRoom class and set it before giving to the listener.
`UserJoinedRoom userJoinedRoom = event.getPayload.getUserJoinedRoom;`
`userJoinedRoom.setInitiator(initiator);`
`listener.onUserJoinedRoom(userJoinedRoom);`
Is it a solution?

Reply:
Don't you think that we could actually use the same strategy for other callbacks?